File: front/front-monopoly/src/services/PlayerService.js
import axios from 'axios';

const PLAYERS = 'http://localhost:9090/api/players';

class PlayerService {
    createPlayer(player) {
        return axios.post(PLAYERS + '/addPlayer', player);
    }

    getPlayerByRoomId(roomId) {
        return axios.get(PLAYERS + '/getPlayerByRoomId/' + roomId);
    }

    getPlayerNextNumberByRoomId(roomId) {
        return axios.get(PLAYERS + '/getPlayerNextNumberByRoomId/' + roomId);
    }

    updatePlayerById(playerId, player) {
        return axios.post(PLAYERS + '/updatePlayerById/' + playerId, player);
    }

    getPlayerById(playerId) {
        return axios.get(PLAYERS + '/getPlayerById/' + playerId);
    }
}

export default new PlayerService();

File: src/main/java/aeh/zzaip/Monopoly/service/PlayerService.java
package aeh.zzaip.Monopoly.service;

import aeh.zzaip.Monopoly.model.Player;

import java.util.List;
import java.util.Optional;

public interface PlayerService {
    List<Player> getAllPlayers();

    Optional<Player> getPlayerById(Long playerId);

    Player addPlayer(Player player);

    Optional<Player> updatePlayerById(Long playerId, Player newPlayer);

    boolean deletePlayerById(Long playerId);

    Optional<List<Player>> getPlayerByRoomId(Long roomId);
}
